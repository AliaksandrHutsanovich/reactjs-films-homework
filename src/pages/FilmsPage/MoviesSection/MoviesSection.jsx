import React, { memo } from 'react';
import { Layout } from 'antd';
import { Tabs } from '../../../components';
import { MovieList, Filter } from '../../../containers';
import { REQUEST_TYPES } from '../../../constants';

import styles from './MoviesSection.scss';

export const renderTrending = () => (<MovieList type={REQUEST_TYPES.TRENDING} />);
export const renderTopRated = () => (<MovieList type={REQUEST_TYPES.TOP_RATED} />);
export const renderComingSoon = () => (<MovieList type={REQUEST_TYPES.COMING_SOON} />);

const tabsConfig = [
  {
    key: '1',
    renderItemChildren: renderTrending,
    tabTitle: 'Trending',
  }, {
    key: '2',
    renderItemChildren: renderTopRated,
    tabTitle: 'Top Rated',
  }, {
    key: '3',
    renderItemChildren: renderComingSoon,
    tabTitle: 'Coming Soon',
  },
];

const MoviesSection = () => (
  <Layout className={styles.wrapper}>
    <Tabs
      tabsConfig={tabsConfig}
      defaultActiveKey="1"
      tabBarExtraContent={<Filter />}
      tabBarGutter={15}
      tabPosition="top"
    />
  </Layout>
);

export default memo(MoviesSection);
