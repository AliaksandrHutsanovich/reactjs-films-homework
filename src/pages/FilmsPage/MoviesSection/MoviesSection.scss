.wrapper {
  padding: 30px 40px;

  @media (max-width: 500px) {
    padding: 30px 0px;
  }
}
