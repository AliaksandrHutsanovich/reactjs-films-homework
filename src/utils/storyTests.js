const storyTests = (stories, renderer) => {
  Object.keys(stories).forEach((storyKey) => {
    if (typeof stories[storyKey] !== 'function') {
      return;
    }
    it(`${storyKey} should be rendered properly`, () => {
      renderer.render(stories[storyKey]());
      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
};

export default storyTests;
