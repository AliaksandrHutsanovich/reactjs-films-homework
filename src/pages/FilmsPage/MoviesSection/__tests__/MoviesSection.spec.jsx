import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import storyTests from '../../../../utils/storyTests';
import MoviesSection, {
  renderTrending,
  renderTopRated,
  renderComingSoon,
} from '../MoviesSection';

jest.mock('../../../../hooks/useDataRequest', () => () => []);

jest.mock('../../../../hooks/useItemsSelector', () => () => {});

jest.mock('../../../../containers/MovieList/MovieList.container', () => () => <div>This is a fake container</div>);

describe('<MoviesSection />', () => {
  const renderer = new ShallowRenderer();

  it('should be rendered properly', () => {
    renderer.render(<MoviesSection />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  storyTests({
    renderTrending,
    renderTopRated,
    renderComingSoon,
  }, renderer);
});
