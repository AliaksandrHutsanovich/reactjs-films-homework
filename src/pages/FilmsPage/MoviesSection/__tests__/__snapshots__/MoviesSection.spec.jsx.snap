// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviesSection /> renderComingSoon should be rendered properly 1`] = `
<div>
  This is a fake container
</div>
`;

exports[`<MoviesSection /> renderTopRated should be rendered properly 1`] = `
<div>
  This is a fake container
</div>
`;

exports[`<MoviesSection /> renderTrending should be rendered properly 1`] = `
<div>
  This is a fake container
</div>
`;

exports[`<MoviesSection /> should be rendered properly 1`] = `
<Layout
  className="wrapper"
>
  <UNDEFINED
    defaultActiveKey="1"
    tabBarExtraContent={<UNDEFINED />}
    tabBarGutter={15}
    tabPosition="top"
    tabsConfig={
      Array [
        Object {
          "key": "1",
          "renderItemChildren": [Function],
          "tabTitle": "Trending",
        },
        Object {
          "key": "2",
          "renderItemChildren": [Function],
          "tabTitle": "Top Rated",
        },
        Object {
          "key": "3",
          "renderItemChildren": [Function],
          "tabTitle": "Coming Soon",
        },
      ]
    }
  />
</Layout>
`;
